Event/RequestListener.php

```diff
@@ -62,45 +69,36 @@ public function __construct(
     public function onKernelRequest(GetResponseEvent $event)
     {
         if (!$event->isMasterRequest()) {
-            // don't do anything if it's not the master request
+            // do nothing if it's not the master request
             return;
         }
 
         $request = $event->getRequest();
 
         if (!$request->isMethod('post')) {
-            // don't do anything if it's not a post request
+            // do nothing if it's not a post request
             return;
         }
 
         try {
             /** @var FormInterface $form */
-            list($formType, $form) = $this->formBuilder->buildByRequest($request);
+            $form = $this->formBuilder->buildByRequest($request);
 
-            if (!$form || !$formType) {
-                // do nothing when no form was found
+            if (!$form || !$form->isSubmitted() || !$form->isValid()) {
+                // do nothing when no form was found or not valid
                 return;
             }
         } catch (\Exception $e) {
             // Catch all exception on build form by request
-
             return;
         }
 
-        if ($form->isSubmitted() && $form->isValid()) {
-            /** @var Dynamic $dynamic */
-            $dynamic = $form->getData();
-            $serializedObject = $dynamic->getForm()->serializeForLocale($dynamic->getLocale(), $dynamic);
-
-            // save
-            $this->formHandler->handle(
-                $form,
-                [
-                    '_form_type' => $formType,
-                    'formEntity' => $serializedObject,
-                ]
-            );
+        /** @var Dynamic $dynamic */
+        $dynamic = $form->getData();
+        $configuration = $this->formConfigurationFactory->buildByDynamic($dynamic);
 
+        if ($this->formHandler->handle($form, $configuration)) {
+            $serializedObject = $dynamic->getForm()->serializeForLocale($dynamic->getLocale(), $dynamic);
             $dynFormSavedEvent = new DynFormSavedEvent($serializedObject, $dynamic);
             $this->eventDispatcher->dispatch(DynFormSavedEvent::NAME, $dynFormSavedEvent);
```

Comment:
The `DynFormSavedEvent` is marked as deprecated. The MailChimpListener is subscribed to it. Should this be refactored to use `FormEvent`?

Reply:
Thx did fix that.